hooks/useElections.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState, useMemo } from "react";
+import { useElectionsQuery, ElectionResult } from "generated";
+
+export function useElections() {
+  const electionsQuery = useElectionsQuery();
+  const { data, isSuccess } = electionsQuery;
+  const [selectedElectionId, setSelectedElectionId] = useState<
+    string | undefined
+  >();
+
+  const elections = useMemo(
+    () =>
+      data
+        ? [...(data?.electionsByUserState as ElectionResult[])].sort((a, b) => {
+            const today = new Date();
+            const distancea = Math.abs(
+              today.getTime() - new Date(a.electionDate).getTime()
+            );
+            const distanceb = Math.abs(
+              today.getTime() - new Date(b.electionDate).getTime()
+            );
+            return distancea - distanceb;
+          })
+        : undefined,
+    [data]
+  );
+
+  useEffect(() => {
+    if (isSuccess && elections)
+      setSelectedElectionId(
+        // Sort by most current election - copy array to preserve chronological order
```

Comment:
Is this comment relevant here anymore? maybe move to above useMemo?

Reply:
sure